src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -123,9 +124,19 @@ internal static void ThrowPlatformNotSupportedException()
 
         internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
         {
+            Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
+
             if (arraySegment.Array == null)
             {
-                throw new ArgumentNullException(parameterName + ".Array");
+                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
+            }
+            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
```

Comment:
I tried getting to this situation with:
```cs
internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
{
    if (arraySegment.Array == null)
    {
        throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
    }
    if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
    {
        throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Offset));
    }
    if (arraySegment.Count < 0 || arraySegment.Count > (arraySegment.Array.Length - arraySegment.Offset))
    {
        throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Count));
    }
}

static void Main(string[] args)
{
    byte[] array = new byte[10];

    ArraySegment<byte> arraySegment = new ArraySegment<byte>(array, 0, 10);
    ValidateArraySegment(arraySegment, nameof(array));

    Array.Resize(ref array, 1);
    ValidateArraySegment(arraySegment, nameof(array));
}
```

but it seems like ArraySegment holds on the old array, not the sized array internal